plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/connection/ConnectionWizardPageModel.java

```diff
@@ -303,8 +303,10 @@ public IStatus getConnectionCreationError() {
 	public IStatus connect() {
 		IStatus status = Status.OK_STATUS;
 		try {
-			if(connection != null) {
-				connection.connect();
+			if(connection != null && !connection.connect()) {
```

Comment:
when connecting many errors can occurr. You can experience network outage, the server can go internal server etc. and last but not least authorization can fail. It it thus imho not sufficient to have a boolean to inidicate whether we could connect or not. We need all details about what went wrong. I thus definitely prefer the prior design where #connect throws, carrying the failure details in the exception.